devicemodel/core/monitor.c

```diff
@@ -465,7 +465,9 @@ void monitor_close(void)
 	/* client buf-mem and fd may be still in use by msg-handler */
 	/* which is handled by mevent */
 	pthread_mutex_lock(&client_mutex);
-	LIST_FOREACH(client, &client_head, list) {
+	list_foreach_safe(client, &client_head, list, pclient) {
+		num_client--;
```

Comment:
num_client will be -- in                 vmm_client_free_res(client);   Do we need this one?